packages/@aws-cdk/aws-ec2/lib/vpc.ts

```diff
@@ -1082,7 +1083,17 @@ export class Vpc extends VpcBase {
       this.vpnGatewayId = vpnGateway.ref;
 
       // Propagate routes on route tables associated with the right subnets
-      const vpnRoutePropagation = props.vpnRoutePropagation || [{ subnetType: SubnetType.PRIVATE }];
+      let vpnRoutePropagation: SubnetSelection[];
```

Comment:
This chain of ifs is not necessary. The default behavior of `selectSubnets` has been changed so that if you don't specify a `subnetType`, it will default to PRIVATE if available and fall back to ISOLATED if not.

So this will do the same:

```ts
const vpnRoutePropagation = props.vpnRoutePropagation ?? [{}];
```

The error message will not be as specific, I suppose, but maybe acceptable?

Reply:
It also falls back to PUBLIC (after ISOLATED), do we want this behavior for VPN route propagation?

Reply:
I'm not sure. How bad is that?

Reply:
It's not bad, it's just weird, if your subnet is PUBLIC you probably don't need a VPN to access it...

Reply:
There are in fact use cases for PUBLIC subnets with VPN access, will update to use the default behavior of `selectSubnets`.